app/models/user.rb

```diff
@@ -9,41 +9,41 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-  has_many :received_invitations, class_name: 'FriendshipInvitation', foreign_key: 'invitee_id'
-  has_many :sent_invitations, class_name: 'FriendshipInvitation', foreign_key: 'inviter_id'
+  has_many :inverted_friendships, class_name: 'Friendship', foreign_key: 'friend_id', dependent: :destroy
+  has_many :friendships, class_name: 'Friendship', foreign_key: 'user_id', dependent: :destroy
 
   def friends
-    friends = received_invitations.map { |invitation| invitation.inviter if invitation.confirmed == true }
-    friends += sent_invitations.map { |invitation| invitation.invitee if invitation.confirmed == true }
-    friends.compact
+    friendships.map { |invitation| invitation.friend if invitation.confirmed == true }.compact
   end
```

Comment:
- [x] we don't need a function to get the list of `friends`. 
```ruby
# User model

has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```
thanks to this, we can find user friends only by user_id in friendships table